sdk/nodejs/automation/stack.ts

```diff
@@ -829,6 +829,99 @@ Event: ${line}\n${e.toString()}`);
         };
     }
 
+    /**
+     * Performs a dry-run destroy of the stack, returning pending changes.
+     *
+     * @param opts
+     *  Options to customize the behavior of the destroy.
+     */
+    async previewDestroy(opts?: DestroyOptions): Promise<PreviewResult> {
+        const args = ["destroy", "--preview-only"];
+        args.push(...this.remoteArgs());
+
+        if (opts) {
+            if (opts.message) {
+                args.push("--message", opts.message);
+            }
+            if (opts.exclude) {
+                for (const eURN of opts.exclude) {
+                    args.push("--exclude", eURN);
+                }
+            }
+            if (opts.target) {
+                for (const tURN of opts.target) {
+                    args.push("--target", tURN);
+                }
+            }
+            if (opts.excludeDependents) {
+                args.push("--exclude-dependents");
+            }
+            if (opts.targetDependents) {
+                args.push("--target-dependents");
+            }
+            if (opts.excludeProtected) {
+                args.push("--exclude-protected");
+            }
+            if (opts.continueOnError) {
+                args.push("--continue-on-error");
+            }
+            if (opts.parallel) {
+                args.push("--parallel", opts.parallel.toString());
+            }
+            if (opts.userAgent) {
+                args.push("--exec-agent", opts.userAgent);
+            }
+            if (opts.refresh) {
+                args.push("--refresh");
+            }
+            if (opts.runProgram !== undefined) {
```

Comment:
For `runProgram`, do we need to set up the inline program similar to here: https://github.com/pulumi/pulumi/blob/master/sdk/nodejs/automation/stack.ts#L776-L799?

Reply:
Good point! Updated the code and added a test